internal/builders/generic/README.md

```diff
@@ -424,3 +425,157 @@ jobs:
       base64-subjects: "${{ needs.build.outputs.hashes }}"
       upload-assets: true # upload to a new release
 ```
+
+### Provenance for Java
+
+If you develop with Java and use [Maven](#maven) or [Gradle](#gradle), you can
+easily generate SLSA3 provenance by updating your existing workflow with the
+steps indicated in the workflow below:
+
+#### Maven
+```
+jobs:
+  build:
+    # ==================================================
+    #
+    # Step 1: Declare an `outputs` for the artifacts generated by
+    # the build and their hashes.
+    #
+    # ==================================================
+    outputs:
+      artifacts: ${{ steps.build.outputs.artifacts }}
+      hashes: ${{ steps.hash.outputs.hashes }}
+
+    [...]
+
+    steps:
+      [...]
+      - name: Build using maven
+        # =================================================
+        #
+        # Step 2: Add an `id: build` field
+        #         to your maven build step.
+        #
+        # =================================================
+        id: build
+        run: |
+          # Your normal build workflow targets here
+          mvn clean package
+          
+          # ======================================================
+          #
+          # Step 3: Save the location of the maven output files
+          #         for easier reference
+          #
+          # =====================================================
+          ARTIFACT_PATTERN=./target/$(mvn help:evaluate -Dexpression=project.artifactId -q -DforceStdout)-$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)*.jar
+          echo "::set-output name=artifact_pattern::$ARTIFACT_PATTERN"
+
+      # ========================================================
+      #
+      # Step 4: Add a step to generate the provenance subjects
+      #         as shown below. Update the sha256 sum arguments
+      #         to include all binaries that you generate
+      #         provenance for.
+      #
+      # ========================================================
+      - name: Generate subject
+        id: hash
+        run: |
+          echo "::set-output name=hashes::$(sha256sum ${{ steps.build.outputs.artifact_pattern }} | base64 -w0)"
+      
+      - name: Upload build artifacts
```

Comment:
his is an optional step correct? Only needed if we want to later upload to the results for the release.
Fine for this PR, just want to be sure.